bin/development-mode

```diff
@@ -0,0 +1,235 @@
+        // nothing to do
+        echo 'Development mode was already disabled.', PHP_EOL;
+        return 0;
+    }
+
+    if (file_exists('config/autoload/development.local.php')) {
+        // optional application config override
+        unlink('config/autoload/development.local.php');
+    }
+
+    unlink('config/development.config.php');
+
+    removeConfigCacheFile(getConfigCacheFile());
+
+    echo 'Development mode is now disabled.', PHP_EOL;
+    return 0;
+}
+
+/**
+ * Enable development mode
+ *
+ * @return int
+ */
+function enable()
+{
+    if (file_exists('config/development.config.php')) {
+        // nothing to do
+        echo 'Already in development mode!', PHP_EOL;
+        return 0;
+    }
+
+    if (! file_exists('config/development.config.php.dist')) {
+        fwrite(
+            STDERR,
+            'MISSING "config/development.config.php.dist". Could not switch to development mode!' . PHP_EOL
+        );
+        return 1;
+    }
+
+    copy('config/development.config.php.dist', 'config/development.config.php');
+
+    if (file_exists('config/autoload/development.local.php.dist')) {
+        // optional application config override
+        copy('config/autoload/development.local.php.dist', 'config/autoload/development.local.php');
+    }
+
+    removeConfigCacheFile(getConfigCacheFile());
+
+    echo 'You are now in development mode.', PHP_EOL;
+    return 0;
+}
+
+/**
+ * Removes the application configuration cache file, if present.
+ */
+function removeConfigCacheFile($configCacheFile)
+{
+    if ($configCacheFile && file_exists($configCacheFile)) {
+        unlink($configCacheFile);
+    }
+}
+
+/**
+ * Retrieve the config cache file, if any.
+ *
+ * @return false|string
+ */
+function getConfigCacheFile()
+{
+    $configCacheDir = getConfigCacheDir();
+    $configCacheKey = getConfigCacheKey();
+
+    if (empty($configCacheDir)) {
+        return false;
+    }
+
+    $path = sprintf('%s/%s.', $configCacheDir, CONFIG_CACHE_BASE);
+
+    if (! empty($configCacheKey)) {
+        $path .= $configCacheKey . '.';
+    }
+
+    return $path . 'php';
+}
+
+/**
+ * Return the configured configuration cache directory, if any.
+ *
+ * @return null|string
+ */
+function getConfigCacheDir()
+{
+    static $dir;
+
+    if ($dir) {
+        return $dir;
+    }
+
+    $config = getApplicationConfig();
+    if (isset($config['module_listener_options']['cache_dir'])
+        && ! empty($config['module_listener_options']['cache_dir'])
+    ) {
+        $dir = $config['module_listener_options']['cache_dir'];
+    }
+
+    return $dir;
+}
+
+/**
+ * Return the configured configuration cache key, if any.
+ *
+ * @return null|string
+ */
+function getConfigCacheKey()
+{
+    static $key;
+
+    if ($key) {
+        return $key;
+    }
+
+    $config = getApplicationConfig();
+    if (isset($config['module_listener_options']['config_cache_key'])
+        && ! empty($config['module_listener_options']['config_cache_key'])
```

Comment:
sorry, `!empty()` definitely will not raise any errors even if the key does not exist
https://3v4l.org/3HtYl

> Determine whether a variable is considered to be empty. A variable is considered empty if it does not exist or if its value equals FALSE. empty() does not generate a warning if the variable does not exist.
http://php.net/empty

Still I think `!empty()` is better choice.